subprojects/core/src/integTest/groovy/org/gradle/caching/internal/packaging/BuildCacheEntryPackingIntegrationTest.groovy

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal.packaging
+
+
+import org.gradle.integtests.fixtures.DirectoryBuildCacheFixture
+import org.gradle.integtests.fixtures.daemon.DaemonIntegrationSpec
+import spock.lang.Issue
+import spock.lang.Unroll
+
+class BuildCacheEntryPackingIntegrationTest extends DaemonIntegrationSpec implements DirectoryBuildCacheFixture {
+    @Issue("https://github.com/gradle/gradle/issues/9877")
+    @Unroll
+    def "can store and load files having #type characters in name (default file encoding #fileEncoding)"() {
```

Comment:
what's the failure reason before the fix?

Reply:
I'm not sure I fully understand your question. The failure that triggered the addition of this test is detailed in the linked issue. There is no fix yet, as I first would like to reproduce the problem. Unfortunately, I couldn't do that yet. That's why the PR is in draft, and in progress.

Does this answer your question? Are you expecting some change in the code?

Reply:
It wasn't clear to me that the problem wasn't reproduced yet, I assumed the detailed PR with the tests meant that at least some of them were failing, otherwise I didn't see what the point of a PR was.
<img width="832" alt="Screen Shot 2019-07-05 at 9 03 49" src="https://user-images.githubusercontent.com/1841944/60704176-fa9fb680-9f03-11e9-9983-c50fce652d74.png">

Reply:
What is the purpose of your comments here? As I said above, this is a draft of a PR, it's not intended to be reviewed yet. I'm just trying to reproduce the problem a user reported, and failing. Nothing to see here yet. :)

Reply:
Ok :)